pyro/contrib/epidemiology/distributions.py

```diff
@@ -38,12 +49,132 @@ def set_approx_sample_thresh(thresh):
         dist.Binomial.approx_sample_thresh = old
 
 
+def _validate_overdispersion(overdispersion):
+    if is_validation_enabled():
+        if not _all(0 <= overdispersion):
+            raise ValueError("Expected overdispersion >= 0")
+        if not _all(overdispersion < 2):
+            raise ValueError("Expected overdispersion < 2")
+
+
+def binomial_dist(total_count, probs, *,
+                  overdispersion=0.):
+    """
+    Returns a Beta-Binomial distribution that is an overdispersed version of a
+    Binomial distribution, according to a parameter ``overdispersion``,
+    typically set in the range 0.1 to 0.5.
+
+    This is useful for (1) fitting real data that is overdispersed relative to
+    a Binomial distribution, and (2) relaxing models of large populations to
+    improve inference. In particular the ``overdispersion`` parameter lower
+    bounds the relative uncertainty in stochastic models such that increasing
+    population leads to a limiting scale-free dynamical system with bounded
+    stochasticity, in contrast to Binomial-based SDEs that converge to
+    deterministic ODEs in the large population limit.
+
+    This parameterization satisfies the following properties:
+
+    1.  Variance increases monotonically in ``overdispersion``.
+    2.  ``overdispersion = 0`` results in a Binomial distribution.
+    3.  ``overdispersion`` lower bounds the relative uncertainty ``std_dev /
+        (total_count * p * q)``, where ``probs = p = 1 - q``, and serves as an
+        asymptote for relative uncertainty as ``total_count → ∞``. This
+        contrasts the Binomial whose relative uncertainty tends to zero.
+    4.  If ``X ~ binomial_dist(n, p, overdispersion=σ)`` then in the large
+        population limit ``n → ∞``, the scaled random variable ``X / n``
+        converges in distribution to ``LogitNormal(log(p/(1-p)), σ)``.
+
+    To achieve these properties we set ``p = probs``, ``q = 1 - p``, and::
+
+        concentration = 1 / (p * q * overdispersion**2) - 1
+
+    :param total_count: Number of Bernoulli trials.
+    :type total_count: int or torch.Tensor
+    :param probs: Event probabilities.
+    :type probs: float or torch.Tensor
+    :param overdispersion: Amount of overdispersion, in the half open interval
+        [0,2). Defaults to zero.
+    :type overdispersion: float or torch.tensor
+    """
+    _validate_overdispersion(overdispersion)
+    if _is_zero(overdispersion):
+        return dist.ExtendedBinomial(total_count, probs)
+    if getattr(probs, "dtype", torch.float64) != torch.float64:
+        warnings.warn("binomial_dist is unstable for dtypes less than torch.float64; "
```

Comment:
is this really the source?

Reply:
Good point. I've moved the warning to `CompartmentalModel.fit()` and `.predict()`.